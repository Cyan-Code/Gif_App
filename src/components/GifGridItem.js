import React from 'react'

export const GifGridItem = ( img ) => {

  return (
    <div className="card animate__fadeIn">
      <p> {img.title} </p>
      <img src={img.url} alt={img.title} />
    </div>
  )
}