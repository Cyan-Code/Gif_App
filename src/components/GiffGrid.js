import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types'

export const GiffGrid = ({ category }) => {

  const {data, loading} = useFetchGifs(category)

  return (
    <>
      <h3 className="card animate__fadeIn"> {category} </h3>
      
      { loading && <p className = "card animate__flash">Loading</p>}

      <div className="card-grid">
        <ol>
          <li>
            {
              data.map ( img => (
                <GifGridItem
                  key = { img.id }
                  { ...img }
                />
              ))
            }
          </li>
        </ol>
      </div>
    </>
  )
}

GiffGrid.propTypes = {
  category: PropTypes.string.isRequired
}
