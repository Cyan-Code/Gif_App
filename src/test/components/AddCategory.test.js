import React from 'react';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas sobre el componente AddCategory', () => {
  
  const setCategories = jest.fn();
  const wrapper = shallow (<AddCategory setCategories={setCategories}/>)

  test('Debe renderizarse de manera correcta', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe de cambiar la caja de texto', () => {

    const input = wrapper.find('input')
    const value = 'hola mundo'
    input.simulate('change', { target: {value} })

  })

  test('No debe de postear la informaciÃ³n onSubmit', () => {

    wrapper.find('form').simulate('submit', { preventDefault(){} })

    expect(setCategories).not.toHaveBeenCalled()

  })
  

  
})


