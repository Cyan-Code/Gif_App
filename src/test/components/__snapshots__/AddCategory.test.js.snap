// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente AddCategory Debe renderizarse de manera correcta 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
